studio/components/interfaces/TableGridEditor/GridHeaderActions.tsx

```diff
@@ -75,62 +86,134 @@ const GridHeaderActions = ({
-          <RenderAPIPreviewToggle />
-        </div>
-      )}
-    </div>
+            </Tooltip.Trigger>
+            <Tooltip.Portal>
+              <Tooltip.Content side="bottom">
+                <Tooltip.Arrow className="radix-tooltip-arrow" />
+                <div
+                  className={[
+                    'rounded bg-scale-100 py-1 px-2 leading-none shadow',
+                    'border border-scale-200',
+                  ].join(' ')}
+                >
+                  <span className="text-xs text-scale-1200">
+                    You need additional permissions to manage your project's data
+                  </span>
+                </div>
+              </Tooltip.Content>
+            </Tooltip.Portal>
+          </Tooltip.Root>
+        )}
+
+        {(table.rls_enabled || showRLSWarning) && (
+          <Link href={`/project/${projectRef}/auth/policies?search=${table.id}`}>
+            <a>
+              <Button
+                type={table.rls_enabled ? 'link' : 'warning'}
+                icon={
+                  table.rls_enabled ? (
+                    <IconLock strokeWidth={2} size={14} />
+                  ) : (
+                    <IconAlertCircle strokeWidth={2} size={14} />
+                  )
+                }
+              >
+                {!table.rls_enabled
+                  ? 'RLS is not enabled'
+                  : `${policies.length == 0 ? 'No' : policies.length} active RLS polic${
+                      policies.length > 1 || policies.length == 0 ? 'ies' : 'y'
+                    }`}
+              </Button>
+            </a>
+          </Link>
+        )}
+
+        <Button
+          type="default"
+          icon={
+            isRealtimeEnabled ? (
+              <IconZap className="text-brand" strokeWidth={1.5} />
+            ) : (
+              <IconZapOff />
+            )
+          }
+          onClick={() => setShowEnableRealtime(true)}
+        >
+          Realtime {isRealtimeEnabled ? 'enabled' : 'disabled'}
+        </Button>
+
+        {doesHaveAutoGeneratedAPIdocs && (
+          <div className="mt-[1px]">
+            <RenderAPIPreviewToggle />
+          </div>
+        )}
+      </div>
+
+      <ConfirmationModal
+        visible={showEnableRealtime}
+        loading={isTogglingRealtime}
+        header={`${isRealtimeEnabled ? 'Disable' : 'Enable'} realtime for ${table.name}`}
+        buttonLabel={`${isRealtimeEnabled ? 'Disable' : 'Enable'} realtime`}
+        buttonLoadingLabel={`${isRealtimeEnabled ? 'Disabling' : 'Enabling'} realtime`}
+        onSelectCancel={() => setShowEnableRealtime(false)}
+        onSelectConfirm={() => toggleRealtime()}
+      >
+        <Modal.Content className="py-4 space-y-2">
+          <p className="text-sm">
+            Any changes on this table will {isRealtimeEnabled && 'no longer '}be broadcasted to
+            authorized subscribers once realtime has been{' '}
+            {isRealtimeEnabled ? 'disabled' : 'enabled'}.
```

Comment:
active voice 😉
(prettier required)
```suggestion
            Once realtime has been {isRealtimeEnabled ? 'disabled' : 'enabled'}, the table will {isRealtimeEnabled && 'no longer '}broadcast any changes to authorized subscribers.
```